src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -345,6 +345,112 @@ void QgsPostgresProvider::setTransaction( QgsTransaction *transaction )
   mTransaction = static_cast<QgsPostgresTransaction *>( transaction );
 }
 
+struct Ewkt
+{
+  int srid = -1;
+  QString wkt;
+};
+
+QgsReferencedGeometry QgsPostgresProvider::fromEwkt( const QString &ewkt, QgsPostgresConn *conn )
+{
+  thread_local const QRegularExpression regularExpressionSRID( "^SRID=(\\d+);" );
+
+  QRegularExpressionMatch regularExpressionMatch = regularExpressionSRID.match( ewkt );
+  if ( !regularExpressionMatch.hasMatch() )
+    return QgsReferencedGeometry();
+
+  Ewkt ewktInfo;
+  ewktInfo.wkt = ewkt.mid( regularExpressionMatch.captured( 0 ).size() );
+  ewktInfo.srid = regularExpressionMatch.captured( 1 ).toInt();
+
+  if ( ewktInfo.srid < 0 )
```

Comment:
I think it's useless since you can't have a negative srid with the regex.

Reply:
Good poit, thanks